### PR TITLE
Fix a bunch of warnings

### PR DESCRIPTION
**Pull request by lhchavez**
This change fixes a bunch of warnings that were discovered by compiling
with `clang -target=i386-pc-linux-gnu`.

### CODE REVIEW

File: src/integer.h [hunk 1]
```diff
@@ -55,16 +55,18 @@ GIT_INLINE(bool) git__add_uint64_overflow(uint64_t *out, uint64_t one, uint64_t
...
 # define git__add_sizet_overflow(out, one, two) \
 	__builtin_uaddl_overflow(one, two, out)
 # define git__multiply_sizet_overflow(out, one, two) \
 	__builtin_umull_overflow(one, two, out)
-#elif (SIZE_MAX == UINT_MAX) && __has_builtin(__builtin_uadd_overflow)
+#elif (__has_builtin(__builtin_add_overflow) || \
+       (defined(__GNUC__) && (__GNUC__ >= 5)))
 # define git__add_sizet_overflow(out, one, two) \
-	__builtin_uadd_overflow(one, two, out)
+	__builtin_add_overflow(one, two, out)
```
Comment (lhchavez) [inline src/integer.h]: This intrinsic handles whatever sizes you throw at it: https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html

But with some odd flag combinations in clang, it requires linking against compiler-rt, which was suboptimal. Keeping the chunk in L58-L63 avoided that case and compiled correctly with the intrinsics in all cases I could think of.

**Comment by ethomson** [comment]
Thanks for doing this - there's a lot of unrelated changes in this first commit, and they're not well described, so I'll need a little bit of time to tease things out (ugh all the different `printf` format descriptors).  It looks correct from first glance, and I know the compiler agrees, but I'd just like to have one more set of eyes on it.

That's not meant as a complaint as much as a warning that I'll need some more time to look at this before I merge this, even though it's a rather small PR.

**Comment by lhchavez** [comment]
Moved the intrinsics part of the change to https://github.com/libgit2/libgit2/pull/4929 to make it easier to review.

**Comment by ethomson** [comment]
Appreciate you bumping the intrinsics out, that's definitely helpful for reviewing.  I think that the printf formatter values need not change to be `I64` - were you seeing a warning from those that I'm not seeing immediately?

**Comment by lhchavez** [comment]
Yes, on [Windows (amd64; MinGW)](https://dev.azure.com/libgit2/libgit2/_build/results?buildId=1173):

```
2019-01-09T01:28:57.4190283Z D:\a\1\s\src\cache.c: In function 'git_cache_dump_stats':
2019-01-09T01:28:57.4190460Z D:\a\1\s\src\cache.c:51:9: warning: format '%u' expects argument of type 'unsigned int', but argument 3 has type 'size_t' {aka 'long long unsigned int'} [-Wformat=]
2019-01-09T01:28:57.4190545Z   printf("Cache %p: %"PRIuZ" items cached, %"PRIdZ" bytes\n",
2019-01-09T01:28:57.4190613Z          ^~~~~~~~~~~~~
2019-01-09T01:28:57.4190657Z    cache, git_cache_size(cache), cache->used_memory);
2019-01-09T01:28:57.4190696Z           ~~~~~~~~~~~~~~~~~~~~~
```

this goes away with the formatting change. let me add a comment.

File: src/cc-compat.h [hunk 2]
```diff
@@ -47,12 +47,23 @@
...
+#		define PRIuZ "I64u"
+#		define PRIxZ "I64x"
+#		define PRIXZ "I64X"
+#		define PRIdZ "I64d"
+#	else
+#		define PRIuZ "Iu"
+#		define PRIxZ "Ix"
+#		define PRIXZ "IX"
+#		define PRIdZ "Id"
+#	endif
```
[the remark this answers is not in the document]
Comment (lhchavez) [reply src/cc-compat.h]: Added a comment.

**Comment by ethomson** [comment]
Cool, thanks for doing this.  Looks good to me.

**Pull request merged**